app/bundles/LeadBundle/Controller/LeadController.php

```diff
@@ -1581,8 +1581,7 @@ public function emailAction($objectId = 0)
                         $user = $this->factory->getUser();
 
                         $mailer->setFrom(
-                            $email['from'],
-                            (strtolower($user->getEmail()) !== strtolower($email['from'])) ? null : $user->getFirstName().' '.$user->getLastName()
+                        $email['from'], ( empty($email['fromname']) == false ) ? $email['fromname'] : ''
```

Comment:
Also, when you are in it, the condition could be simplified from
`( empty($email['fromname']) == false ) ? $email['fromname'] : ''`
to
`empty($email['fromname']) ? '' : $email['fromname']`